docs/README.md

```diff
@@ -75,6 +108,281 @@ The Sail-operator does not manage Gateways. You can deploy a gateway manually ei
+
+#### Example using the InPlace strategy
+
+Prerequisites:
+* sail-operator needs to be installed.
+
+Steps:
+1. Create the `istio-system` and the `istio-cni` namespaces.
+
+    ```bash
+    kubectl create namespace istio-system
+    kubectl create namespace istio-cni
+    ```
+
+2. Create the `Istio-cni` resource
+  
+      ```bash
+      cat <<EOF | kubectl apply -f -
+      apiVersion: operator.istio.io/v1alpha1
+      kind: IstioCNI
+      metadata:
+        name: default
+      spec:
+        version: v1.21.0
+        namespace: istio-cni
+      EOF
+      ```
+
+3. Create the `Istio` resource
+
+    ```bash
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: InPlace
+        inactiveRevisionDeletionGracePeriodSeconds: 30
+      version: v1.21.0
+    EOF
+    ```
+
+4. Confirm the installation and version of the control plane.
+
+    ```bash
+    $ kubectl get istio -n istio-system
+    NAME      READY   STATUS    IN USE   VERSION   AGE
+    default   True    Healthy   True     v1.21.0   2m
+    ```
+
+5. Create namespace `bookinfo` and deploy bookinfo application
+
+    ```bash
+    kubectl create namespace bookinfo
+    kubectl label namespace bookinfo istio.io/rev=default
+    kubectl apply -n bookinfo -f https://raw.githubusercontent.com/istio/istio/release-1.22/samples/bookinfo/platform/kube/bookinfo.yaml
+    ```
+
+6. Perform the update of the control plane by changing the version in the Istio resource
+
+    ```bash
+    cat <<EOF | kubectl apply -f-
+    apiVersion: operator.istio.io/v1alpha1
+    kind: Istio
+    metadata:
+      name: default
+    spec:
+      namespace: istio-system
+      updateStrategy:
+        type: InPlace
+        inactiveRevisionDeletionGracePeriodSeconds: 30
+      version: v1.21.2
+    EOF
+    ```
+
+7. Confirm the istio resource version was updated
+
+    ```bash
+    kubectl get istio -n istio-system
+    NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   VERSION   AGE
+    default   1           1       1        Healthy           v1.21.2   12m
+    ```
+
+8. Delete `bookinfo` pod to trigger sidecar injection with the new version
+
+    ```bash
+    kubectl delete pods --all -n bookinfo
+    ```
+
+9. Confirm that the new version is used in the sidecar
+
+    ```bash
+    kubectl get pod -n bookinfo -l app=details -o yaml | yq '.items[].spec.containers[].image'
+    docker.io/istio/examples-bookinfo-details-v1:1.19.1
+    docker.io/istio/proxyv2:1.21.2
+    ```
+
+### RevisionBased
+When the `RevisionBased` strategy is used, a new Istio control plane instance is created for every change to the Istio.spec.version field. The old control plane remains in place until all workloads have been moved to the new control plane instance, then, the old control plane is deleted.
```

Comment:
Is the old control plane removed automatically?

Reply:
Yes, it will be removed automatically when all the pods are moved to the new one